src/persist-client/src/internal/metrics.rs

```diff
@@ -571,6 +571,10 @@ pub struct BatchWriteMetrics {
     pub(crate) goodbytes: IntCounter,
     pub(crate) seconds: Counter,
     pub(crate) write_stalls: IntCounter,
+
+    pub(crate) step_consolidation: Counter,
+    pub(crate) step_columnar_encoding: Counter,
+    pub(crate) step_part_writing: Counter,
```

Comment:
don't love adding the metrics here compared to `CompactionStepTimings` (which I find easier to use with how they're parameterized), but I didn't want to burn too many cycles on refactoring the metrics structure and overcomplicating this PR